src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/VM.java

```diff
@@ -502,14 +502,13 @@ private VM(TypeDataBase db, JVMDebugger debugger, boolean isBigEndian) {
     boolType = (CIntegerType) db.lookupType("bool");
 
     minObjAlignmentInBytes = getObjectAlignmentInBytes();
-    if (minObjAlignmentInBytes == 8) {
-      logMinObjAlignmentInBytes = 3;
-    } else if (minObjAlignmentInBytes == 16) {
-      logMinObjAlignmentInBytes = 4;
-    } else {
-      throw new RuntimeException("Object alignment " + minObjAlignmentInBytes + " not yet supported");
+    if (minObjAlignmentInBytes < 8 || minObjAlignmentInBytes > 256 ||
```

Comment:
Is it necessary to restrict it to the 8 -> 256 range. I know this is the current hotspot restriction, but if hotspot decided to allow a larger range, is there any reason SA wouldn't still work?

Reply:
Are you proposing to drop the range check altogether? I can do that.

Reply:
You could keep in the check to make sure it is aligned to a power of 2, since I doubt SA would work with a value that isn't, but then I don't see the possibility of hotspot ever supporting it either.